contracts/BufferedToken.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./WadMath.sol";
+import "./oracles/IOracle.sol";
+
+/**
+ * @title Buffered Token
+ * @author Alex Roan (@alexroan)
+ * @notice This contract represents a token which accepts ETH as collateral to mint and burn ERC20 tokens.
+ * It's functions abstract all oracle, conversion, debt ratio and ETH buffer calculations, so that
+ * inheriting contracts can focus on their minting and burning algorithms.
+ */
+contract BufferedToken is ERC20 {
```

Comment:
The original purpose from @alexroan was to separate the code and make it more manageable. It worked well in that sense for testing.

As it stands, `BufferedToken` is a general buffered token, we are choosing the underlying to be Eth, but the code could be reused to have anything else as the underlying.

We can choose to merge them, but I would probably wait just until we have tests for `fund` and `defund`.

Reply:
Yeah, it's for the separation of concerns.  `EthPool`, `debtRatio()`, oracle conversions, etc are managed by BufferedToken so that USM could focus solely on the mathematics of `mint`, `burn`, `fund` and `defund`.

I'm hesitant to merge the two.

Reply:
I'm now starting to test `fund()`, and I don't think the split works, actually. Both `fund()` and `defund()` manipulate `ethPool`, which is a `BufferedToken` variable. It's not great to manipulate parent variables from inheriting classes.

I'll try to finish the testing as-is, then we merge this PR into `master`, and then I think we should merge `BufferedToken` and `USM` into a single contract (in a separate PR), separating concerns just by using internal and public functions (business logic + api).

Reply:
OK, at least the conceptual distinction between "a pool of [some unspecified asset]" and "a subtype which is specifically a pool of ETH" makes sense to me now.  But, there are complexity pros & cons each way...  My instinct would be to only introduce that separation if we're ever actually trying to build a pool with a different asset.  Anyway yeah as @albertocuestacanada said let's carry on with the current structure and consider merging once we're back on master.  (Unless the split becomes a more immediate impediment to getting tests to pass etc.)